src/platform/Darwin/PlatformManagerImpl.cpp

```diff
@@ -53,6 +55,40 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack()
     return err;
 }
 
+CHIP_ERROR PlatformManagerImpl::_StartEventLoopTask()
+{
+    if (mIsWorkQueueRunning == false)
+    {
+        mIsWorkQueueRunning = true;
+        dispatch_resume(mWorkQueue);
+    }
+
+    return CHIP_NO_ERROR;
+};
+
+CHIP_ERROR PlatformManagerImpl::_StopEventLoopTask()
+{
+
+    if (mIsWorkQueueRunning == true)
+    {
+        mIsWorkQueueRunning = false;
+
+        // dispatch_sync is used in order to guarantee serialization of the caller with
+        // respect to any tasks that might already be on the queue, or running.
+        dispatch_sync(mWorkQueue, ^{
```

Comment:
For better or worse, I had thought that the posix platform was discontinuing execution regardless of whether additional events were enqueued.  Which behavior is correct?  This or that?

Also, given that events can post other events, is this potentially futile?

Reply:
Answered below.  Yes, this is slightly different behavior, but not in a substantive way.  This is only delaying suspension of processing as an artifact of the need to serialize suspension of event processing on this platform within the event stream itself.  In all of our platforms, we can still have remaining events enqueued when we suspend processing.